packages/connect/src/protocol/invoke-implementation.ts

```diff
@@ -111,6 +114,7 @@ export function transformInvokeImplementation<
             return {
               stream: true,
               message: output,
+              method: spec.method as DescMethodStreaming<I, O>,
```

Comment:
Defining the `DescMethod*` types as an intersection (e.g. `DescMethod & { methodKind: "unary" }`) doesn't limit the union of `methodKind` in all contexts:

```ts
function f(x: DescMethodUnary<any, any>) {
  x.methodKind; // "unary" | "server_streaming" | "client_streaming" | "bidi_streaming"
}
```

I think we can replace the property instead (`Omit<DescMethod, "methodKind"> & {methodKind: "unary"}`). We shouldn't need the cast here anymore, and users won't need it either in similar contexts.

Not sure if this also applies to `input` and `output`.